tests/foreman/ui/test_rhcloud_insights.py

```diff
@@ -215,3 +217,221 @@ def test_host_sorting_based_on_recommendation_count():
+
+    :expectedresults:
+        1. rake command deletes insights reporting status of host.
+
+    :BZ: 1962930
+
+    :CaseAutomation: Automated
+    """
+    org, ak = organization_ak_setup
+    rhel7_contenthost.configure_rhai_client(
+        satellite=rhcloud_sat_host, activation_key=ak.name, org=org.label, rhel_distro=DISTRO_RHEL7
+    )
+    with Session(hostname=rhcloud_sat_host.hostname) as session:
+        session.organization.select(org_name=org.name)
+        session.location.select(loc_name=DEFAULT_LOC)
+        values = session.host.get_details(rhel7_contenthost.hostname)
+        assert values['properties']['properties_table']['Insights'] == 'Reporting clear'
+        values = session.host.get_details(rhel8_insights_vm.hostname)
+        assert values['properties']['properties_table']['Insights'] == 'Reporting clear'
+        # Clean insights status
+        result = rhcloud_sat_host.run(
+            f'foreman-rake rh_cloud_insights:clean_statuses SEARCH="{rhel7_contenthost.hostname}"'
+        )
+        assert 'Deleted 1 insights statuses' in result.stdout
+        assert result.status == 0
+        values = session.host.get_details(rhel7_contenthost.hostname)
+        with pytest.raises(KeyError):
+            values['properties']['properties_table']['Insights']
+        values = session.host.get_details(rhel8_insights_vm.hostname)
+        assert values['properties']['properties_table']['Insights'] == 'Reporting clear'
+        result = rhel7_contenthost.run('insights-client')
+        assert result.status == 0
+        values = session.host.get_details(rhel7_contenthost.hostname)
+        assert values['properties']['properties_table']['Insights'] == 'Reporting clear'
+
+
+@pytest.mark.run_in_one_thread
+@pytest.mark.tier2
+def test_delete_host_having_insights_recommendation(
+    rhel8_contenthost,
+    organization_ak_setup,
+    set_rh_cloud_token,
+    rhcloud_sat_host,
+):
+    """Verify that host having insights recommendations can be deleted from Satellite.
+
+    :id: 07914ff7-e230-4416-8664-7d357e9966f3
+
+    :customerscenario: true
+
+    :Steps:
+        1. Prepare misconfigured machine and upload its data to Insights
+        2. Add Cloud API key in Satellite
+        3. Sync insights recommendations.
+        4. Sync RH Cloud inventory status.
+        5. Go to Hosts -> All Hosts
+        6. Assert there is "Recommendations" column containing insights recommendation count.
+        7. Try to delete host.
+
+    :expectedresults:
+        1. host having insights recommendations is deleted from Satellite.
+
+    :CaseImportance: Critical
+
+    :BZ: 1860422, 1928652
+
+    :CaseAutomation: Automated
+    """
+    org, ak = organization_ak_setup
+    rhel8_contenthost.configure_rhai_client(
+        satellite=rhcloud_sat_host, activation_key=ak.name, org=org.label, rhel_distro=DISTRO_RHEL8
+    )
+    rhel8_contenthost.run('dnf update -y dnf')
+    rhel8_contenthost.run('sed -i -e "/^best/d" /etc/dnf/dnf.conf')
```

Comment:
```suggestion
    rhel8_contenthost.run('sed -i -e "/^best/d" /etc/dnf/dnf.conf')
```
What does this do? Would it be worth a one-line comment?

Reply:
It helps create dnf related insights recommendation for rhel8 host which is required for proper testing.
updated the pr.